third_party/terraform/tests/resource_bigquery_table_test.go

```diff
@@ -471,12 +471,135 @@ func TestUnitBigQueryDataTable_jsonEquivalency(t *testing.T) {
 	}
 }
 
+func TestUnitBigQueryDataTable_schemaIsChangable(t *testing.T) {
+	t.Parallel()
+	for _, testcase := range testUnitBigQueryDataTableIsChangableTestCases {
+		testcase.check(t)
+		testcaseNested := &testUnitBigQueryDataTableJSONChangeableTestCase{
+			testcase.name + "Nested",
+			fmt.Sprintf("[{\"name\": \"someValue\", \"type\" : \"INTEGER\", \"fields\" : %s }]", testcase.jsonOld),
+			fmt.Sprintf("[{\"name\": \"someValue\", \"type\" : \"INT64\", \"fields\" : %s }]", testcase.jsonNew),
+			testcase.changeable,
+		}
+		testcaseNested.check(t)
+	}
+}
+
 type testUnitBigQueryDataTableJSONEquivalencyTestCase struct {
 	jsonA      string
 	jsonB      string
 	equivalent bool
 }
 
+type testUnitBigQueryDataTableJSONChangeableTestCase struct {
+	name       string
+	jsonOld    string
+	jsonNew    string
+	changeable bool
+}
+
+func (testcase *testUnitBigQueryDataTableJSONChangeableTestCase) check(t *testing.T) {
+	var old, new interface{}
+	if err := json.Unmarshal([]byte(testcase.jsonOld), &old); err != nil {
+		panic(fmt.Sprintf("unable to unmarshal json - %v", err))
+	}
+	if err := json.Unmarshal([]byte(testcase.jsonNew), &new); err != nil {
+		panic(fmt.Sprintf("unable to unmarshal json - %v", err))
+	}
+	changeable, err := resourceBigQueryTableSchemaIsChangeable(old, new)
+	if err != nil {
+		t.Errorf("ahhhh an error I did not expect this! especially not on testscase %s - %s", testcase.name, err)
+	}
+	if changeable != testcase.changeable {
+		t.Errorf("expected changeable result of %v but got %v for testcase %s", testcase.changeable, changeable, testcase.name)
+	}
+
+	d := &ResourceDiffMock{
+		Before: map[string]interface{}{},
+		After:  map[string]interface{}{},
+	}
+
+	d.Before["schema"] = testcase.jsonOld
+	d.After["schema"] = testcase.jsonNew
+
+	err = resourceBigQueryTableSchemaCustomizeDiffFunc(d)
+	if err != nil {
+		t.Errorf("error on testcase %s - %w", testcase.name, err)
+	}
+	if !testcase.changeable != d.IsForceNew {
+		t.Errorf("%s: expected d.IsForceNew to be %v, but was %v", testcase.name, !testcase.changeable, d.IsForceNew)
+	}
+}
+
+var testUnitBigQueryDataTableIsChangableTestCases = []testUnitBigQueryDataTableJSONChangeableTestCase{
+	{
+		"defaultEquality",
+		"[{\"name\": \"someValue\", \"type\" : \"INTEGER\", \"mode\" : \"NULLABLE\", \"description\" : \"someVal\" }]",
+		"[{\"name\": \"someValue\", \"type\" : \"INTEGER\", \"mode\" : \"NULLABLE\", \"description\" : \"someVal\" }]",
+		true,
```

Comment:
One approach I like to make these kinds of tests more readable is naming constants. For example `const IsChangeable bool = true` (and conversely, `IsNotChangeable` for false)
```suggestion
		IsChangeable,
```

Reply:
going for a different approach of naming the parameters... hoping this mades it more readable. I feel weird injecting something as generic constant such as IsChangeable into the google package namespace.